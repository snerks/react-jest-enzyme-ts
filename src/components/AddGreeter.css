.AddGreeter {
  margin: 20px;
  padding: 20px;
  border: 2px solid #00FFD8;
  background: #DDFFEE;
  text-align: center;
}
