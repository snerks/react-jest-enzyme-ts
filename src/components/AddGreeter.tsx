import * as React from 'react';
import './AddGreeter.css';

export interface AddGreeterProps {
    addGreeting: Function;
}

export interface AddGreeterState {
    greetingName: string;
}

class AddGreeter extends React.Component<AddGreeterProps, AddGreeterState> {
    constructor(props: AddGreeterProps) {
        super(props);

        this.state = { greetingName: '' };
    }

    // handleUpdate(event: React.ChangeEvent<HTMLInputElement>) {
    //     this.setState({ greetingName: event!.target!.value });
    // }

    // tslint:disable-next-line:no-any
    handleUpdate(event: any) {
        this.setState({ greetingName: event!.target!.value });
    }

    addGreeting() {
        this.props.addGreeting(this.state.greetingName);
        this.setState({ greetingName: '' });
    }

    render() {
        return (
            <div className="AddGreeter">
                <input
                    type="text"
                    value={this.state.greetingName}
                    onChange={(e) => this.handleUpdate(e)}
                />
                &nbsp;&nbsp;
                <button onClick={() => this.addGreeting()}>Add</button>
            </div>
        );
    }
}

export default AddGreeter;
