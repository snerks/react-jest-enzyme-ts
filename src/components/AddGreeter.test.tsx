import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

import AddGreeter from './AddGreeter';

describe('AddGreeter', () => {
    const mockAddGreeting = jest.fn();
    const _component = shallow(
        <AddGreeter addGreeting={mockAddGreeting} />
    );

    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            // tslint:disable-next-line:no-empty
            <AddGreeter addGreeting={() => { }} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains the form', () => {
        expect(_component.find('input')).toHaveLength(1);
        expect(_component.find('button')).toHaveLength(1);
    });

    it('calls the passed in addGreeting function when add button is clicked', () => {
        _component.find('button').simulate('click');
        expect(mockAddGreeting).toBeCalled();
    });

    it('updates the form when keys are pressed', () => {
        const updateKey = 'foo';
        (_component.instance() as AddGreeter).handleUpdate({
            target: {
                value: updateKey
            }
        });

        expect(_component.state('greetingName')).toEqual(updateKey);
    });

    it('blanks out the greetingName when the button is clicked', () => {
        const updateKey = 'a';

        (_component.instance() as AddGreeter).handleUpdate({ target: { value: updateKey } });
        expect(_component.state('greetingName')).toEqual(updateKey);

        _component.find('button').simulate('click');
        expect(_component.state('greetingName')).toHaveLength(0);
    });

});