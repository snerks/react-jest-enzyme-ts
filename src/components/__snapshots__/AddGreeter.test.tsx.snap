exports[`AddGreeter renders and matches our snapshot 1`] = `
<div
  className="AddGreeter">
  <input
    onChange={[Function]}
    type="text"
    value="" />
    
  <button
    onClick={[Function]}>
    Add
  </button>
</div>
`;
